### PR TITLE
Finish profile page right side

### PR DESCRIPTION
**Pull request by mattBman23**

### CODE REVIEW

File: client/components/profilePage/myProfile.jsx [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
 import UserBookClubs from './userBookClubs.jsx';
 import AllUsersModal from './allUsersModal.jsx';
 import AllFriendsModal from './allFriendsModal.jsx';
-import { Button, ListGroup, Container } from 'react-bootstrap';
+import { Button, ListGroup, Container, Badge, Row, Col } from 'react-bootstrap';
 import AuthContext from '../../context/authContext.jsx';
 
-export default function myFriends({user = {friends:[]}}) {
+export default function myFriends({ user = { friends: [] } }) {
   const currentUser = {
```
Comment (daniel-he-dev) [inline client/components/profilePage/myProfile.jsx:12]: Are we going with static data for now?

Comment (daniel-he-dev) [reply client/components/profilePage/myProfile.jsx:12]: nvm i see you switched it out

File: client/context/context.jsx [hunk 2]
```diff
@@ -5,13 +5,34 @@ import axios from 'axios';
 export const AppContext = createContext();
 
 export const AppProvider = ({ children }) => {
+<<<<<<< HEAD
```
Comment (daniel-he-dev) [inline client/context/context.jsx:8]: Please resolve this conflict. You may not have pulled the latest version of the code

File: client/context/authContext.jsx [hunk 3]
```diff
@@ -7,6 +7,7 @@ export const AuthContext = createContext();
 export const AuthProvider = ({ children }) => {
   const checkAuth = async () => {
     let userData = await axios.get('/checkauth');
+    localStorage.setItem('usrID', JSON.stringify(userData.data._id));
```
Comment (daniel-he-dev) [inline client/context/authContext.jsx:10]: does this get cleaned up on logout somewhere? Need to make sure there is not user id that stays in localstorage if the user is logged out.

**Review by daniel-he-dev — CHANGES_REQUESTED** [review]

File: client/components/profilePage/friendRecommendations.jsx [hunk 4]
```diff
@@ -1,32 +1,40 @@
...
-                <p>{friend.recommendation.summary}</p>
-                <p>{friend.recommendation.description}}</p>
+
+      <Accordion defaultActiveKey='0'>
+        {reviewList.length > 0 &&
+          reviewList.map((friend, i) => {
+            return (
+              <Card key={friend._id}>
+                <Card.Header>
+                  <Accordion.Toggle as={Button} variant='link' eventKey={i + 1}>
```
Comment (jfleming9357) [inline client/components/profilePage/friendRecommendations.jsx:24]: Don't think it really matters, but why is the +1 needed? Does i alone not work?

File: client/components/profilePage/main.jsx [hunk 5]
```diff
@@ -1,13 +1,14 @@
 import MyLibrary from './myLibrary.jsx';
 import MyProfile from './myProfile.jsx';
-import React from 'react';
+import React, { useContext } from 'react';
 import css from './profileStyles.css';
+import { AuthContext } from '../../context/authContext.jsx';
```
Comment (jfleming9357) [inline client/components/profilePage/main.jsx:5]: Context looks unused? Assuming it didn't work or something?

**Review by daniel-he-dev — APPROVED** [review]

File: client/components/profilePage/userBookClubs.jsx [hunk 6]
```diff
@@ -4,30 +4,48 @@ import { AppContext } from '../../context/context.jsx';
 import { AuthContext } from '../../context/authContext.jsx';
 import { Card, Col, Row, Tab, Tabs } from 'react-bootstrap';
 import './profileStyles.css';
+import axios from 'axios';
+import { Link } from 'react-router-dom';
 
-export default function UserBookClubs({user = {bookclubs: []}}) {
+export default function UserBookClubs({ user = { bookclubs: [] } }) {
+  const usrId = JSON.parse(localStorage.getItem('usrID')) || '';
```
Comment (jfleming9357) [inline client/components/profilePage/userBookClubs.jsx:11]: The way around not using localstorage is to conditionally run the axios call in the below use effect based on whether user is defined, and to have the use effect watch the user parameter

File: client/context/authContext.jsx (hunk 3 above)
Comment (jfleming9357) [reply client/context/authContext.jsx:10]: Agreed with daniel, this either needs to be cleaned up on logout, or follow my suggestion on the comment I made regarding your useEffect function

File: server/routes/UserRoute.js [hunk 7]
```diff
@@ -381,4 +380,18 @@ router.get('/userclubs/:id', async (req, res) => {
   }
 });
 
+// @desc    User  get all current user book clubs
+// @route   GET /user/userclubs/:id
+// @access  Private
+router.post('/bookclubs', async (req, res) => {
```
Comment (jfleming9357) [inline server/routes/UserRoute.js:386]: Looks like the @route needs to be fixed. Though I personally think that was the better way to do it, considering you are getting data and not posting it.

**Review by jfleming9357 — CHANGES_REQUESTED** [review]

**Review by jfleming9357 — APPROVED** [review]

File: client/components/profilePage/friendRecommendations.jsx (hunk 4 above)
Comment (mattBman23) [reply client/components/profilePage/friendRecommendations.jsx:24]: nope, it seems Accordion starts at 1 and not 0

File: client/components/profilePage/main.jsx (hunk 5 above)
Comment (mattBman23) [reply client/components/profilePage/main.jsx:5]: Was experimenting, we are getting user from props so dont need context.

File: server/routes/UserRoute.js (hunk 7 above)
Comment (mattBman23) [reply server/routes/UserRoute.js:386]: Will fix it

**Pull request merged**